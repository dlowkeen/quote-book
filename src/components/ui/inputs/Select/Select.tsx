import React from 'react';

import * as styles from '../inputs.css';

interface ISelectProps {
  options: string[];
  label: string;
  onChange: (e: any) => void;
  value: string;
}

const Select: React.FunctionComponent<ISelectProps> = ({
  options,
  label,
  onChange,
  value,
}) => {
  const optionSource = options || [];

  const mappedOptions = optionSource.map((o: any, i: number) => {
    const option = {
      label: '',
      value: '',
    };

    if (typeof o === 'string') {
      option.label = o;
      option.value = o;
    } else {
      option.label = o.label;
      option.value = o.value;
    }

    return (
      <option key={i} value={option.value}>
        {' '}
        {option.label}{' '}
      </option>
    );
  });

  return (
    <div className={`${styles.formElement} ${styles.formElementSelect}`}>
      <label>{label}</label>
      <select value={value} onChange={onChange}>
        {mappedOptions}
      </select>
    </div>
  );
};

export default Select;
