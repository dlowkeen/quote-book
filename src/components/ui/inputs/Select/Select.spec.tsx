import React from 'react';
import renderer from 'react-test-renderer';

import Select from './Select';

describe('The Select component', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(
        <Select
          options={['this', 'that', 'the other']}
          value='3'
          label='label'
          onChange={() => 'changed!'}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
