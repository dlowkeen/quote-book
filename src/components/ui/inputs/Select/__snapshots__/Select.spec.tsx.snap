// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Select component renders as expected 1`] = `
<div
  className="inputs__form-element__VT4Ts inputs__form-element-select___TUE5"
>
  <label>
    label
  </label>
  <select
    onChange={[Function]}
    value="3"
  >
    <option
      value="this"
    >
       
      this
       
    </option>
    <option
      value="that"
    >
       
      that
       
    </option>
    <option
      value="the other"
    >
       
      the other
       
    </option>
  </select>
</div>
`;
